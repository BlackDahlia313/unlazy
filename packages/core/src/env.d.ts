declare global {
  const __UNLAZY_HASH_DECODING__: boolean | undefined
  const __UNLAZY_LOGGING__: boolean | undefined
}

export {}
