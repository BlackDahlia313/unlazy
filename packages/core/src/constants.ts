export const DEFAULT_BLURHASH_SIZE = 32
