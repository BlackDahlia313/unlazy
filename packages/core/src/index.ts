export { lazyLoad, autoSizes, loadImage } from './lazyLoad'
