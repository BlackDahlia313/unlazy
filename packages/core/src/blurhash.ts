import { decodeBlurHash } from 'fast-blurhash'
import { DEFAULT_PLACEHOLDER_SIZE } from './constants'
import { rgbaToDataUri } from './utils/dataUri'

export interface BlurHashOptions {
  /**
   * Aspect ratio (width / height) of the BlurHash image to be decoded.
   *
   * @default 1 (square aspect ratio)
   */
  ratio?: number

  /**
   * The size of the longer edge (width or height) of the BlurHash image to be
   * decoded, depending on the aspect ratio.
   *
   * @default 32
   */
  size?: number
}

export function createPngDataUri(
  hash: string,
  {
    ratio = 1,
    size = DEFAULT_PLACEHOLDER_SIZE,
  }: BlurHashOptions = {},
) {
  const { width, height } = getAspectRatioDimensions(ratio, size)
  const rgba = decodeBlurHash(hash, width, height)
  return rgbaToDataUri(width, height, rgba)
}

function getAspectRatioDimensions(ratio: number, size: number) {
  const isLandscapeOrSquare = ratio >= 1

  return {
    width: isLandscapeOrSquare ? size : Math.round(size * ratio),
    height: isLandscapeOrSquare ? Math.round(size / ratio) : size,
  }
}
