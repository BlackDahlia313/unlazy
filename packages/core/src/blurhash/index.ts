import { decodeBlurHash } from 'fast-blurhash'
import { DEFAULT_BLURHASH_SIZE } from '../constants'
import { getBlurhashDimensions, isSSR } from '../utils'
import { encodeSvgAsDataUri, generateBlurredImageSvg } from '../svg'
import type { UnLazyLoadOptions } from '../types'
import { getPngFromRgbaArr } from './png'

export type BlurhashOptions = {
  /**
   * Aspect ratio (width / height) of the decoded BlurHash image.
   *
   * @default 1 (square aspect ratio)
   */
  ratio?: number
} & Pick<UnLazyLoadOptions, 'blurhashSize'>

export function createPngDataUriFromBlurHash(
  hash: string,
  {
    ratio = 1,
    blurhashSize = DEFAULT_BLURHASH_SIZE,
  }: BlurhashOptions = {},
) {
  const { width, height } = getBlurhashDimensions(ratio, blurhashSize)
  const rgba = decodeBlurHash(hash, width, height)
  const png = getPngFromRgbaArr(rgba, width, height)
  let base64: string

  if (isSSR) {
    const pngArray = new Uint8Array(Array.from(png, c => c.charCodeAt(0)))
    // eslint-disable-next-line n/prefer-global/buffer
    base64 = Buffer.from(pngArray).toString('base64')
  }
  else {
    base64 = btoa(png)
  }

  return `data:image/png;base64,${base64}`
}

export function createSvgDataUriFromBlurHash(
  hash: string,
  {
    ratio = 1,
    blurhashSize = DEFAULT_BLURHASH_SIZE,
  }: BlurhashOptions = {},
) {
  const pngDataUri = createPngDataUriFromBlurHash(hash, { ratio, blurhashSize })
  const { width, height } = getBlurhashDimensions(ratio, blurhashSize)
  const svg = generateBlurredImageSvg(pngDataUri, width, height)

  return encodeSvgAsDataUri(svg)
}
