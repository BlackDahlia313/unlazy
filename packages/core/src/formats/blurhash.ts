import { decodeBlurHash } from 'fast-blurhash'
import { DEFAULT_PLACEHOLDER_SIZE } from '../constants'
import { getScaledDimensions } from '../utils'
import { rgbaToDataUri } from '../utils/dataUri'

export interface BlurHashOptions {
  /**
   * Aspect ratio (width / height) of the BlurHash image to be decoded.
   *
   * @default 1 (square aspect ratio)
   */
  ratio?: number

  /**
   * The size of the longer edge (width or height) of the BlurHash image to be
   * decoded, depending on the aspect ratio.
   *
   * @default 32
   */
  size?: number
}

export function createPngDataUri(
  hash: string,
  {
    ratio = 1,
    size = DEFAULT_PLACEHOLDER_SIZE,
  }: BlurHashOptions = {},
) {
  const { width, height } = getScaledDimensions(ratio, size)
  const rgba = decodeBlurHash(hash, width, height)
  return rgbaToDataUri(width, height, rgba)
}
