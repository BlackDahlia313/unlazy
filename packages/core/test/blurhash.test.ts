import { describe, expect, test } from 'vitest'
import { createPngDataUriFromBlurHash, createSvgDataUriFromBlurHash } from '../src/blurhash'

describe('createSvgDataUriFromBlurHash', () => {
  test('should return a valid data URL', () => {
    const hash = 'LKO2:N%2Tw=w]~RBVZRi};RPxuwH'
    const dataURL = createSvgDataUriFromBlurHash(hash)
    expect(dataURL).toMatchSnapshot()
  })
})

describe('createPngDataUriFromBlurHash', () => {
  test ('should return a valid data URL', () => {
    const hash = 'LKO2:N%2Tw=w]~RBVZRi};RPxuwH'
    const dataURL = createPngDataUriFromBlurHash(hash)
    expect(dataURL).toMatchSnapshot()
  })
})
