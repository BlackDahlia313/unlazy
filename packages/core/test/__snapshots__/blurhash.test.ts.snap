// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createPngDataUriFromBlurHash > should return a valid data URL 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAQK0lEQVR4AQEgEN/vAPzBof/7waL/+sCi//jAo//1v6T/8r6m/++9p//su6j/6bqo/+e5qP/luKj/5bim/+a3pf/nt6L/6beg/+y4nf/vuJr/8rmX//W7lf/3vJP/+L6T//m/lP/5wZX/+MOY//bEm//0xp//8cej/+7Jp//ryqr/6Mqt/+bLsP/ky7H/APvBof/7waL/+sCi//i/o//1v6T/8r6m/++8p//ru6j/6bqo/+a5qP/luKj/5bem/+W3pf/nt6L/6beg/+y3nf/vuJr/8rmX//S7lf/3vJT/+L6T//m/lP/4wZX/98OY//bEm//zxp//8Mej/+3Ip//qyar/58qt/+XLr//ky7H/APrAov/6wKL/+cCj//a/pP/0vqX/8b2m/+68p//qu6j/6Lqo/+a5qP/kuKf/5Lem/+W3pP/mtqL/6Leg/+u3nf/uuJr/8bmX//O6lf/1u5T/972U//i/lP/3wJb/9sKY//TEm//yxZ//78aj/+zHpv/pyKr/5sms/+TKr//jyrD/APjAov/4wKP/97+j//W/pP/yvqX/772m/+y7p//puqj/5rmo/+S4qP/jt6f/47am/+O2pP/ltqL/57ag/+q2nf/st5r/77iY//K5lv/0u5X/9byU//a+lf/1v5b/9MGY//PCm//wxJ//7cWi/+rGpv/nx6n/5Mer/+LIrf/gyK//APa/o//1v6T/9L+k//K+pf/wvab/7bym/+q7p//nuaj/5Lio/+K3qP/htqf/4bWm/+G1pP/jtaL/5bWg/+i1nf/qtpv/7beY//C4l//xupX/87uV//O8lf/zvpf/8r+Z//DBm//twp7/6sOi/+fEpf/kxaj/4cWq/9/GrP/exq3/APK+pP/yvqX/8b6l/++9pf/tvKb/6run/+e6p//kuKj/4reo/+C2p//ftaf/37Sl/9+zpP/hs6L/47Og/+W0nf/otZv/6raZ/+23mP/vuJf/8LmW//C7lv/wvJf/7r6Z/+y/m//qwJ7/58Gh/+TBpP/gwqb/3sKo/9vDqv/aw6v/AO++pv/uvab/7b2m/+u8pv/pu6f/5rqn/+S4p//ht6j/37an/920p//cs6b/3LKl/9yyo//esqL/4LKg/+Kynv/ls5z/57Sa/+q1mf/rtpj/7LiX/+y5mP/supj/6rua/+i8nP/mvZ7/476g/9++ov/cv6T/2b+m/9e/p//WwKj/AOq9p//qvKf/6byn/+e7p//luqj/47mo/+C3qP/etaf/27Sn/9qzpv/Zsqb/2bGk/9qwo//bsKH/3bCg/9+xnv/isZz/5LKb/+azmv/ntZn/6LaZ/+i3mf/ouJn/5rma/+S6nP/hup3/3ruf/9u7of/Xu6L/1byk/9K8pf/RvKb/AOa8qf/lvKn/5Lup/+O6qf/huaj/3reo/9y2qP/atKf/2LKn/9axpv/WsKX/1q+k/9auo//YrqH/2q6g/9yvnv/esJ3/4LCc/+Kym//js5r/5LSa/+S1mv/jtpr/4bab/9+3nP/ct53/2bee/9a4n//SuKH/z7ih/824ov/MuKP/AOG7qv/hu6r/4Lqq/965qv/cuKn/2rap/9i0qP/Ws6f/1LGm/9Ovpv/SrqT/0q2j/9Osov/UrKH/1qyg/9itn//arp7/3K6d/96vnP/fsZz/4LKb/9+ym//es5v/3LSc/9q0nP/XtJ3/1LSd/9C0nv/NtJ//yrSf/8ezoP/Gs6D/ANy6rP/cuqz/27mr/9q4q//Yt6r/1rWp/9SzqP/Ssaf/0K+m/8+upf/PrKT/z6uj/8+qov/RqqH/0qqg/9Srn//Wq57/2Kye/9qtnf/brp3/26+d/9uwnf/ZsZz/17Gc/9WxnP/SsZ3/zrGd/8uwnf/HsJ3/xK+d/8Kvnf/Ar53/ANi6rf/Xua3/1rmt/9W4rP/Ttqv/0rSq/9Cyqf/OsKf/zK6m/8uspf/LqqP/y6mi/8yoof/NqKD/z6ig/9Gpn//TqZ//1Kqe/9arnv/WrJ7/162e/9aunv/Vrp7/0q6d/9Cunf/Mrpz/ya2c/8WtnP/CrJv/vqub/7yrmv+6q5r/ANO5r//Tua//0riu/9G3rf/Ptaz/zrOr/8yxqf/Kr6f/yaym/8iqpP/HqaP/yKeh/8mmoP/KpqD/y6af/82nn//Pp5//0aif/9Kpn//Sqp//0quf/9Krn//QrJ//zqye/8urnf/Hq5z/xKqb/8Cpm/+8qJr/uaiZ/7anmP+1p5j/AM+5sP/PubD/zriw/823rv/Mta3/yrOr/8iwqf/Hrqf/xaul/8WppP/Ep6L/xaah/8WloP/HpJ//yKSf/8qln//MpZ//zaag/86noP/PqKD/zqmg/86poP/Mqp//yqmf/8epnv/DqJ3/v6eb/7ummv+3pZn/tKSX/7Gkl/+wo5b/AMy6sv/MubL/y7ix/8q3sP/Jta7/x7Os/8Wwqv/Eraf/wqql/8Koo//BpqH/wqSg/8Kjn//Eop//xaKf/8ejn//JpJ//yqWg/8umof/LpqH/y6eh/8qoof/IqKD/xqif/8Onnv+/pp3/u6Wb/7ekmv+zo5j/sKKX/62hlf+roJX/AMq6s//KurP/ybmy/8i3sf/Gta//xLOs/8Owqv/Braf/wKql/7+no/+/paH/v6Og/8Cin//BoZ7/w6Ge/8Shn//Gop//yKOg/8mkof/JpaH/yaai/8imov/GpqH/w6ag/8Cln/+8pJ3/uKOb/7Simv+voZj/rKCW/6mflf+nnpT/AMi7tP/Iu7T/x7qz/8a4sf/Ftq//w7Ot/8Gwqv+/raj/vqml/72no/+9pKH/vaKf/76hnv+/oJ7/waCe/8Ogn//EoZ//xqKg/8ejof/HpKL/x6Si/8alov/EpaL/waWh/76koP+6o57/taKc/7Ghmv+toJj/qZ6W/6aelf+knZT/AMi9tf/IvLX/x7u0/8W5sv/Et7D/wrSt/8Cwq/++raj/vaql/7ymov+7pKD/vKGf/7ygnv++n53/v56d/8Gfnv/Dn5//xKCg/8Wiof/Go6L/xqSi/8Wko//DpKL/wKSh/72koP+5o5//tKKd/7Chm/+rn5n/qJ6X/6Sdlv+jnZX/AMi+tv/Ivrb/x721/8a7s//EuLH/wrWu/8Cxq/++rqj/vKql/7umov+6o6D/u6Ge/7ufnf+9np3/vp6d/8Cenv/Cn5//xKCg/8Whof/GoqL/xaOj/8Sko//DpKP/wKSi/7ykof+4o5//tKKe/6+hnP+roJr/p5+Y/6Sel/+inZb/AMrAt//JwLf/yL61/8e8tP/FubH/wrau/8Cyq/++rqj/vKql/7unov+6o6D/uqGe/7ufnf+8nZz/vp2c/8Cdnf/Cnp7/xJ+f/8Wgof/GoqL/xqOj/8Wko//DpKP/waSj/72kov+5pKD/taOf/7Cinf+roZz/p6Ca/6Sfmf+inpj/AMzCuP/Lwrf/ysC2/8m+tP/Gu7L/xLiv/8G0rP+/sKj/vaul/7unov+6pJ//uqGd/7ufnP+8nZv/vp2c/8CdnP/CnZ3/xJ+f/8agoP/HoaH/x6Oi/8ako//FpaP/wqWj/7+lov+7paL/tqSg/7Gjn/+to57/qaKc/6Whm/+joZv/AM/EuP/OxLj/zcK2/8vAtf/JvbL/xrmv/8O1rP/Asaj/vqyl/7yoov+7pJ//uqGd/7ufnP+8nZv/v5yb/8Gdm//DnZ3/xp6e/8ign//JoqH/yaOi/8mko//HpaP/xaaj/8Gmo/+9pqP/uaai/7Smof+vpaD/q6Sf/6iknv+lo53/ANLGuP/Rxrj/0MS3/87Ctf/Lv7L/yLuv/8W3rP/Csqn/v66l/72pov+8pZ//u6Kd/7yfm/+9nZr/v5ya/8Kdm//FnZz/x5+d/8mgnv/LoqD/y6Sh/8ulov/Kp6P/x6ek/8SopP/AqKT/vKij/7eoo/+yqKL/rqei/6qnof+op6H/ANbJuf/VyLj/1Me3/9HEtf/OwbP/y72w/8e5rP/EtKn/wa+l/76qov+9pp//vKOd/72gm/++npr/wZ2Z/8Sdmv/GnZv/yZ+c/8ygnf/Nop//zqSh/86mov/NqKP/y6mk/8eqpf/DqqX/v6ul/7qrpf+1q6X/saqk/66qpP+sqqT/ANnLuf/Zyrj/18m3/9XGtf/Sw7P/zr+w/8q7rf/Gtqn/w7Gl/8Csov++p5//vqOc/76gmv/Anpn/wp2Z/8Wdmf/Inpr/y5+b/86hnP/Qo57/0aWg/9Gnov/QqaP/zquk/8uspf/Hrab/wq2n/76up/+5rqf/ta6n/7Gup/+vrqf/AN3Nuf/dzbj/28u3/9jItv/VxbP/0cGw/828rf/It6n/xbKm/8Ktov/AqJ//v6Sc/7+hmv/Bn5n/xJ6Y/8edmP/Knpn/zqCa/9Ghm//TpJ3/1Kaf/9Soof/Tq6P/0ayl/86upv/Lr6f/xrCo/8Gxqf+8sar/uLGq/7Wxqv+zsar/AOHPuf/gz7j/3s23/9zKtv/Yx7P/1MOw/8++rf/Luan/x7Om/8Ouov/BqZ//wKWc/8Gimv/Dn5j/xZ6X/8mel//Mn5j/0KCZ/9Oim//VpZz/16ef/9eqof/XrKP/1a6l/9Kwp//Osaj/yrKq/8Wzq//AtKz/vLSt/7i0rf+2tK3/AOTRuf/j0Lj/4s+3/9/Mtv/bybP/1sSw/9K/rf/Nuqr/ybWm/8Wvov/Dqp//wqac/8Kimv/EoJj/x5+X/8qel//On5f/0qGY/9Wjmv/YpZz/2qie/9qroP/araP/2LCl/9Wyp//Rs6n/zbWr/8i2rf/Dt67/v7ev/7u3sP+5uLD/AOfSuf/m0rj/5NC3/+HOtv/dyrP/2cax/9TBrf/Pu6r/yram/8ewo//Eq5//w6ec/8Ojmv/FoJj/yJ+W/8yflv/Qn5b/1KGX/9ejmf/appv/3Kmd/92soP/crqP/2rGl/9izqP/Utar/0Les/8u4rv/GubD/wrmx/766sv+8urL/AOnUuf/o07j/5tG3/+PPtv/fy7P/2sex/9XCrf/QvKr/zLem/8ixo//FrJ//xKec/8Skmf/GoZf/yZ+W/82flf/RoJb/1aGX/9mkmP/cppr/3qmd/9+sn//er6L/3bKl/9q1qP/Wt6v/0rit/826r//Iu7H/xLuz/8C8tP++vLT/AOvUuf/q1Lj/6NK3/+XQtv/hzLP/3Mix/9fCrf/Rvar/zbem/8myo//GrJ//xaic/8Wkmf/HoZf/yqCW/86flf/SoJX/1qKW/9qkmP/dp5r/36qc/+Ctn//gsKL/3rOl/9y1qP/YuKv/07mu/8+7sP/KvLL/xr20/8K+tf/Avrb/AOzVuf/r1Lj/6dO3/+bQtv/izLP/3cix/9fDrv/Svar/zbim/8myo//HrZ//xaic/8akmf/HoZf/yqCW/86glf/ToJX/16KW/9ukl//ep5n/4Kqc/+Gtn//hsaL/37Ol/922qf/ZuKz/1Lqu/9C8sf/LvbP/x761/8O+tv/Bv7f/kJaUV1R52cIAAAAASUVORK5CYII="`;

exports[`createSvgDataUriFromBlurHash > should return a valid data URL 1`] = `"data:image/svg+xml,%3Csvg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%2032.00%2032.00%22%3E%3Cfilter%20id=%22a%22%20color-interpolation-filters=%22sRGB%22%3E%3CfeGaussianBlur%20stdDeviation=%22.2%22%3E%3C/feGaussianBlur%3E%3CfeComponentTransfer%3E%3CfeFuncA%20type=%22discrete%22%20tableValues=%221%201%22%3E%3C/feFuncA%3E%3C/feComponentTransfer%3E%3C/filter%3E%3Cimage%20filter=%22url(%23a)%22%20x=%220%22%20y=%220%22%20width=%22100%25%22%20height=%22100%25%22%20href=%22data:image/png%3Bbase64%2CiVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAQK0lEQVR4AQEgEN/vAPzBof/7waL/%2BsCi//jAo//1v6T/8r6m/%2B%2B9p//su6j/6bqo/%2Be5qP/luKj/5bim/%2Ba3pf/nt6L/6beg/%2By4nf/vuJr/8rmX//W7lf/3vJP/%2BL6T//m/lP/5wZX/%2BMOY//bEm//0xp//8cej/%2B7Jp//ryqr/6Mqt/%2BbLsP/ky7H/APvBof/7waL/%2BsCi//i/o//1v6T/8r6m/%2B%2B8p//ru6j/6bqo/%2Ba5qP/luKj/5bem/%2BW3pf/nt6L/6beg/%2By3nf/vuJr/8rmX//S7lf/3vJT/%2BL6T//m/lP/4wZX/98OY//bEm//zxp//8Mej/%2B3Ip//qyar/58qt/%2BXLr//ky7H/APrAov/6wKL/%2BcCj//a/pP/0vqX/8b2m/%2B68p//qu6j/6Lqo/%2Ba5qP/kuKf/5Lem/%2BW3pP/mtqL/6Leg/%2Bu3nf/uuJr/8bmX//O6lf/1u5T/972U//i/lP/3wJb/9sKY//TEm//yxZ//78aj/%2BzHpv/pyKr/5sms/%2BTKr//jyrD/APjAov/4wKP/97%2Bj//W/pP/yvqX/772m/%2By7p//puqj/5rmo/%2BS4qP/jt6f/47am/%2BO2pP/ltqL/57ag/%2Bq2nf/st5r/77iY//K5lv/0u5X/9byU//a%2Blf/1v5b/9MGY//PCm//wxJ//7cWi/%2BrGpv/nx6n/5Mer/%2BLIrf/gyK//APa/o//1v6T/9L%2Bk//K%2Bpf/wvab/7bym/%2Bq7p//nuaj/5Lio/%2BK3qP/htqf/4bWm/%2BG1pP/jtaL/5bWg/%2Bi1nf/qtpv/7beY//C4l//xupX/87uV//O8lf/zvpf/8r%2BZ//DBm//twp7/6sOi/%2BfEpf/kxaj/4cWq/9/GrP/exq3/APK%2BpP/yvqX/8b6l/%2B%2B9pf/tvKb/6run/%2Be6p//kuKj/4reo/%2BC2p//ftaf/37Sl/9%2BzpP/hs6L/47Og/%2BW0nf/otZv/6raZ/%2B23mP/vuJf/8LmW//C7lv/wvJf/7r6Z/%2By/m//qwJ7/58Gh/%2BTBpP/gwqb/3sKo/9vDqv/aw6v/AO%2B%2Bpv/uvab/7b2m/%2Bu8pv/pu6f/5rqn/%2BS4p//ht6j/37an/920p//cs6b/3LKl/9yyo//esqL/4LKg/%2BKynv/ls5z/57Sa/%2Bq1mf/rtpj/7LiX/%2By5mP/supj/6rua/%2Bi8nP/mvZ7/476g/9%2B%2Bov/cv6T/2b%2Bm/9e/p//WwKj/AOq9p//qvKf/6byn/%2Be7p//luqj/47mo/%2BC3qP/etaf/27Sn/9qzpv/Zsqb/2bGk/9qwo//bsKH/3bCg/9%2Bxnv/isZz/5LKb/%2Bazmv/ntZn/6LaZ/%2Bi3mf/ouJn/5rma/%2BS6nP/hup3/3ruf/9u7of/Xu6L/1byk/9K8pf/RvKb/AOa8qf/lvKn/5Lup/%2BO6qf/huaj/3reo/9y2qP/atKf/2LKn/9axpv/WsKX/1q%2Bk/9auo//YrqH/2q6g/9yvnv/esJ3/4LCc/%2BKym//js5r/5LSa/%2BS1mv/jtpr/4bab/9%2B3nP/ct53/2bee/9a4n//SuKH/z7ih/824ov/MuKP/AOG7qv/hu6r/4Lqq/965qv/cuKn/2rap/9i0qP/Ws6f/1LGm/9Ovpv/SrqT/0q2j/9Osov/UrKH/1qyg/9itn//arp7/3K6d/96vnP/fsZz/4LKb/9%2Bym//es5v/3LSc/9q0nP/XtJ3/1LSd/9C0nv/NtJ//yrSf/8ezoP/Gs6D/ANy6rP/cuqz/27mr/9q4q//Yt6r/1rWp/9SzqP/Ssaf/0K%2Bm/8%2Bupf/PrKT/z6uj/8%2Bqov/RqqH/0qqg/9Srn//Wq57/2Kye/9qtnf/brp3/26%2Bd/9uwnf/ZsZz/17Gc/9WxnP/SsZ3/zrGd/8uwnf/HsJ3/xK%2Bd/8Kvnf/Ar53/ANi6rf/Xua3/1rmt/9W4rP/Ttqv/0rSq/9Cyqf/OsKf/zK6m/8uspf/LqqP/y6mi/8yoof/NqKD/z6ig/9Gpn//TqZ//1Kqe/9arnv/WrJ7/162e/9aunv/Vrp7/0q6d/9Cunf/Mrpz/ya2c/8WtnP/CrJv/vqub/7yrmv%2B6q5r/ANO5r//Tua//0riu/9G3rf/Ptaz/zrOr/8yxqf/Kr6f/yaym/8iqpP/HqaP/yKeh/8mmoP/KpqD/y6af/82nn//Pp5//0aif/9Kpn//Sqp//0quf/9Krn//QrJ//zqye/8urnf/Hq5z/xKqb/8Cpm/%2B8qJr/uaiZ/7anmP%2B1p5j/AM%2B5sP/PubD/zriw/823rv/Mta3/yrOr/8iwqf/Hrqf/xaul/8WppP/Ep6L/xaah/8WloP/HpJ//yKSf/8qln//MpZ//zaag/86noP/PqKD/zqmg/86poP/Mqp//yqmf/8epnv/DqJ3/v6eb/7ummv%2B3pZn/tKSX/7Gkl/%2Bwo5b/AMy6sv/MubL/y7ix/8q3sP/Jta7/x7Os/8Wwqv/Eraf/wqql/8Koo//BpqH/wqSg/8Kjn//Eop//xaKf/8ejn//JpJ//yqWg/8umof/LpqH/y6eh/8qoof/IqKD/xqif/8Onnv%2B/pp3/u6Wb/7ekmv%2Bzo5j/sKKX/62hlf%2BroJX/AMq6s//KurP/ybmy/8i3sf/Gta//xLOs/8Owqv/Braf/wKql/7%2Bno/%2B/paH/v6Og/8Cin//BoZ7/w6Ge/8Shn//Gop//yKOg/8mkof/JpaH/yaai/8imov/GpqH/w6ag/8Cln/%2B8pJ3/uKOb/7Simv%2BvoZj/rKCW/6mflf%2BnnpT/AMi7tP/Iu7T/x7qz/8a4sf/Ftq//w7Ot/8Gwqv%2B/raj/vqml/72no/%2B9pKH/vaKf/76hnv%2B/oJ7/waCe/8Ogn//EoZ//xqKg/8ejof/HpKL/x6Si/8alov/EpaL/waWh/76koP%2B6o57/taKc/7Ghmv%2BtoJj/qZ6W/6aelf%2BknZT/AMi9tf/IvLX/x7u0/8W5sv/Et7D/wrSt/8Cwq/%2B%2Braj/vaql/7ymov%2B7pKD/vKGf/7ygnv%2B%2Bn53/v56d/8Gfnv/Dn5//xKCg/8Wiof/Go6L/xqSi/8Wko//DpKL/wKSh/72koP%2B5o5//tKKd/7Chm/%2Brn5n/qJ6X/6Sdlv%2BjnZX/AMi%2Btv/Ivrb/x721/8a7s//EuLH/wrWu/8Cxq/%2B%2Brqj/vKql/7umov%2B6o6D/u6Ge/7ufnf%2B9np3/vp6d/8Cenv/Cn5//xKCg/8Whof/GoqL/xaOj/8Sko//DpKP/wKSi/7ykof%2B4o5//tKKe/6%2BhnP%2BroJr/p5%2BY/6Sel/%2BinZb/AMrAt//JwLf/yL61/8e8tP/FubH/wrau/8Cyq/%2B%2Brqj/vKql/7unov%2B6o6D/uqGe/7ufnf%2B8nZz/vp2c/8Cdnf/Cnp7/xJ%2Bf/8Wgof/GoqL/xqOj/8Wko//DpKP/waSj/72kov%2B5pKD/taOf/7Cinf%2BroZz/p6Ca/6Sfmf%2Binpj/AMzCuP/Lwrf/ysC2/8m%2BtP/Gu7L/xLiv/8G0rP%2B/sKj/vaul/7unov%2B6pJ//uqGd/7ufnP%2B8nZv/vp2c/8CdnP/CnZ3/xJ%2Bf/8agoP/HoaH/x6Oi/8ako//FpaP/wqWj/7%2Blov%2B7paL/tqSg/7Gjn/%2Bto57/qaKc/6Whm/%2BjoZv/AM/EuP/OxLj/zcK2/8vAtf/JvbL/xrmv/8O1rP/Asaj/vqyl/7yoov%2B7pJ//uqGd/7ufnP%2B8nZv/v5yb/8Gdm//DnZ3/xp6e/8ign//JoqH/yaOi/8mko//HpaP/xaaj/8Gmo/%2B9pqP/uaai/7Smof%2BvpaD/q6Sf/6iknv%2Blo53/ANLGuP/Rxrj/0MS3/87Ctf/Lv7L/yLuv/8W3rP/Csqn/v66l/72pov%2B8pZ//u6Kd/7yfm/%2B9nZr/v5ya/8Kdm//FnZz/x5%2Bd/8mgnv/LoqD/y6Sh/8ulov/Kp6P/x6ek/8SopP/AqKT/vKij/7eoo/%2ByqKL/rqei/6qnof%2Bop6H/ANbJuf/VyLj/1Me3/9HEtf/OwbP/y72w/8e5rP/EtKn/wa%2Bl/76qov%2B9pp//vKOd/72gm/%2B%2Bnpr/wZ2Z/8Sdmv/GnZv/yZ%2Bc/8ygnf/Nop//zqSh/86mov/NqKP/y6mk/8eqpf/DqqX/v6ul/7qrpf%2B1q6X/saqk/66qpP%2BsqqT/ANnLuf/Zyrj/18m3/9XGtf/Sw7P/zr%2Bw/8q7rf/Gtqn/w7Gl/8Csov%2B%2Bp5//vqOc/76gmv/Anpn/wp2Z/8Wdmf/Inpr/y5%2Bb/86hnP/Qo57/0aWg/9Gnov/QqaP/zquk/8uspf/Hrab/wq2n/76up/%2B5rqf/ta6n/7Gup/%2Bvrqf/AN3Nuf/dzbj/28u3/9jItv/VxbP/0cGw/828rf/It6n/xbKm/8Ktov/AqJ//v6Sc/7%2Bhmv/Bn5n/xJ6Y/8edmP/Knpn/zqCa/9Ghm//TpJ3/1Kaf/9Soof/Tq6P/0ayl/86upv/Lr6f/xrCo/8Gxqf%2B8sar/uLGq/7Wxqv%2Bzsar/AOHPuf/gz7j/3s23/9zKtv/Yx7P/1MOw/8%2B%2Brf/Luan/x7Om/8Ouov/BqZ//wKWc/8Gimv/Dn5j/xZ6X/8mel//Mn5j/0KCZ/9Oim//VpZz/16ef/9eqof/XrKP/1a6l/9Kwp//Osaj/yrKq/8Wzq//AtKz/vLSt/7i0rf%2B2tK3/AOTRuf/j0Lj/4s%2B3/9/Mtv/bybP/1sSw/9K/rf/Nuqr/ybWm/8Wvov/Dqp//wqac/8Kimv/EoJj/x5%2BX/8qel//On5f/0qGY/9Wjmv/YpZz/2qie/9qroP/araP/2LCl/9Wyp//Rs6n/zbWr/8i2rf/Dt67/v7ev/7u3sP%2B5uLD/AOfSuf/m0rj/5NC3/%2BHOtv/dyrP/2cax/9TBrf/Pu6r/yram/8ewo//Eq5//w6ec/8Ojmv/FoJj/yJ%2BW/8yflv/Qn5b/1KGX/9ejmf/appv/3Kmd/92soP/crqP/2rGl/9izqP/Utar/0Les/8u4rv/GubD/wrmx/766sv%2B8urL/AOnUuf/o07j/5tG3/%2BPPtv/fy7P/2sex/9XCrf/QvKr/zLem/8ixo//FrJ//xKec/8Skmf/GoZf/yZ%2BW/82flf/RoJb/1aGX/9mkmP/cppr/3qmd/9%2Bsn//er6L/3bKl/9q1qP/Wt6v/0rit/826r//Iu7H/xLuz/8C8tP%2B%2BvLT/AOvUuf/q1Lj/6NK3/%2BXQtv/hzLP/3Mix/9fCrf/Rvar/zbem/8myo//GrJ//xaic/8Wkmf/HoZf/yqCW/86flf/SoJX/1qKW/9qkmP/dp5r/36qc/%2BCtn//gsKL/3rOl/9y1qP/YuKv/07mu/8%2B7sP/KvLL/xr20/8K%2Btf/Avrb/AOzVuf/r1Lj/6dO3/%2BbQtv/izLP/3cix/9fDrv/Svar/zbim/8myo//HrZ//xaic/8akmf/HoZf/yqCW/86glf/ToJX/16KW/9ukl//ep5n/4Kqc/%2BGtn//hsaL/37Ol/922qf/ZuKz/1Lqu/9C8sf/LvbP/x761/8O%2Btv/Bv7f/kJaUV1R52cIAAAAASUVORK5CYII=%22%3E%3C/image%3E%3C/svg%3E"`;
